### ISSUE TITLE
importccl: slice bounds out of range panic

### ISSUE THREAD

**Issue by irfansharif** [cockroachdb/cockroach#58649]
I've seen this panic on a couple of my open PRs now:
```
BenchmarkPgCopyConvertRecord
panic: runtime error: slice bounds out of range [1:0]

goroutine 12029 [running]:
github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc.buildWritableNonPrimary(0xc000fbda00, 0xc001486000, 0x3, 0xc00220b000, 0x8330eb4c)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc/index.go:470 +0x265
github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc.(*indexCache).cachedIndexes(0xc000fbda00, 0xc000fbda98, 0x4ae3bd0, 0x3, 0xc001486000, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc/index.go:420 +0xc2
github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc.(*indexCache).writableNonPrimaryIndexes(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc/index.go:480
github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc.(*wrapper).WritableNonPrimaryIndexes(0xc001486000, 0x2, 0xc004b00000, 0xc000157b01)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc/table_desc.go:323 +0x65
github.com/cockroachdb/cockroach/pkg/sql/row.MakeInserter(0x50e8d00, 0xc003266000, 0x0, 0xc00036c000, 0xc00036c000, 0x0, 0xc001486000, 0xc009794800, 0x10, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/inserter.go:53 +0x86
github.com/cockroachdb/cockroach/pkg/sql/row.NewDatumRowConverter(0x50e8d00, 0xc003266000, 0xc001486000, 0x0, 0x0, 0x0, 0xc0003f6d80, 0xc00010ed80, 0x0, 0xc002f13c20, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/row_converter.go:336 +0x4fe
github.com/cockroachdb/cockroach/pkg/ccl/importccl.makeDatumConverter(0x50e8d00, 0xc003266000, 0xc008d522d0, 0xc000fce080, 0xc0035ea674, 0xc00320c300, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:449 +0x85
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*parallelImporter).importWorker(0xc01e820000, 0x50e8d00, 0xc003266000, 0xf, 0x50558e0, 0xc001ce4000, 0xc008d522d0, 0xc000fce080, 0xc0002d8100, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:653 +0x6a
github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1.1(0x50e8d00, 0xc003266000, 0xf, 0xc0048c76a0, 0xffffffffffffffff)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:553 +0x91
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1(0x50e8d00, 0xc003266000, 0xc0050d1620, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0x42
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0x0, 0xc0047a07c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0024800c0, 0xc002480540)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```

[Here](https://teamcity.cockroachdb.com/viewLog.html?buildId=2568352&buildTypeId=Cockroach_UnitTests)' is the most recent failure.

**Comment by irfansharif** [comment]
This is a pretty persistent error it seems (just hit it [again](https://teamcity.cockroachdb.com/viewLog.html?buildId=2568408&buildTypeId=Cockroach_UnitTests)).

@postamar, @dt: Was there a recent change that introduced the instability? If so, perhaps it's worthwhile to temporarily revert it? If not, I'd be happy to rubberstamp a PR that skips this benchmark (if that's the only one that's affected) in the interest of unblocking PRs.

**Comment by dt** [comment]
I don't know any of the details here and would defer to sql-schema, but just guessing based on what I've seen fly by in my inbox I think a revert already merged in #58642 ?

**Comment by postamar** [comment]
Indeed, I broke `master` yesterday evening and merged a revert a few minutes ago. Sorry for the trouble 😬
